src/Builder/FrontendDialectTransformer.cpp

```diff
@@ -1406,21 +1410,36 @@ int ImportFrontendModelArray(const void *onnxBuffer, int size,
 }
 
 // Return 0 on success, error otherwise.
-int ImportFrontendModelFile(std::string model_fname, MLIRContext &context,
+int ImportFrontendModelFile(StringRef model_fname, MLIRContext &context,
     OwningOpRef<ModuleOp> &module, std::string *errorMessage,
     ImportOptions options) {
   onnx::ModelProto model;
-  std::fstream input(model_fname, std::ios::in | std::ios::binary);
-  // check if the input file is opened
-  if (!input.is_open()) {
-    *errorMessage = "Unable to open or access " + model_fname;
-    return InvalidInputFileAccess;
-  }
+  if (model_fname == "-" || model_fname.endswith(".json")) {
+    auto buf = openInputFile(model_fname, errorMessage);
+    if (!buf) {
+      return InvalidInputFileAccess;
+    }
+    google::protobuf::StringPiece json(
```

Comment:
idea: we could remove `//` comments from buf here before calling JsonStringToMessage, then we wouldn't need to strip the comments with sed in the lit tests